pkg/app/app.go

```diff
@@ -282,8 +281,16 @@ backup_spec = EXCLUDED.backup_spec`
 			if err != nil {
 				return int64(0), errors.Wrap(err, "failed to marshal diff")
 			}
-
 			diffSummary = string(b)
+
+			// check if version needs additional configuration
```

Comment:
Can we pull this out of the function? so that the function does only what it claims?  UpdateConfigValuesInDB() wouldn't expect to also have business logic that calculates the status of each downstream to me. What options can make this clearer and more testable?